### PR TITLE
ref(py3): Add default cache key prefix for PY3

### PR DESCRIPTION
**Pull request by BYK**
Supercedes getsentry/getsentry#4711, depends on #22322.
getsentry/getsentry#4711 needs to be reverted after this lands.

### CODE REVIEW

File: src/sentry/runner/initializer.py [hunk 1]
```diff
@@ -283,6 +274,17 @@ def configure_structlog():
...
+    # Just reuse the integration app for Single Org / Self-Hosted as
+    # it doesn't make much sense to use 2 separate apps for SSO and
+    # integration.
+    if settings.SENTRY_SINGLE_ORGANIZATION:
+        options_mapper.update(
+            {
+                "github-app.client-id": "GITHUB_APP_ID",
+                "github-app.client-secret": "GITHUB_API_SECRET",
+            }
+        )
```
Comment (BYK) [inline src/sentry/runner/initializer.py:286]: This change is to align it with this file in general. Can move it to a separate PR if you think that's better.

Comment (EvanPurkhiser) [reply src/sentry/runner/initializer.py:286]: I would prefer this in a separate PR yeah

**Review by evanpurkhiser — CHANGES_REQUESTED** [review]

File: src/sentry/runner/initializer.py [hunk 2]
```diff
@@ -314,6 +314,9 @@ def initialize_app(config, skip_service_validation=False):
         settings.CSRF_COOKIE_PATH = getattr(settings, "SESSION_COOKIE_PATH", "/")
 
     settings.CACHES["default"]["VERSION"] = settings.CACHE_VERSION
+    for key in settings.CACHES:
+        if not hasattr(settings.CACHES[key], "KEY_PREFIX"):
+            settings.CACHES[key]["KEY_PREFIX"] = "v0:" if six.PY3 else ""
```
Comment (mattrobenolt) [inline src/sentry/runner/initializer.py:319]: Why don't we do this:

```python
if six.PY3:
    for key in settings.CACHES:
        settings.CACHES[key]["VERSION"] = 2
```

There are 2 options relevant here, the KEY_PREFIX, and the VERSION. We really only wanna bump the version then we don't make the key names longer (which use more memory). The version defaults to 1 already, so we just bump this value instead.

https://github.com/django/django/blob/stable/1.11.x/django/core/cache/backends/base.py#L27-L35

**Review by mattrobenolt — CHANGES_REQUESTED** [review]

**Comment by mattrobenolt** [comment]
I see that you're mirroring the logic in getsentry, but we should change that to bump the `VERSION` value instead there too.

Right now keys already have a version placeholder, and the keys are generated as such: `:1:foo`. With this change, we generate a key name such as `v0::1:foo` adding 3 bytes to each key name, when we ultimately just need to bump the `1` to a `2` instead.

See also: https://docs.djangoproject.com/en/1.11/topics/cache/#cache-versioning

**Comment by BYK** [comment]
Sounds great to me. I'll submit a patch to getsentry/getsentry and change this one. Thanks for the great suggestion.

@EvanPurkhiser @wedamija assuming this is okay by you too as it achieves the same result with less hacks and friction.

@mattrobenolt thinking again, do we need an extra logic in getsentry/getsentry at all? I'll just remove it as it will be obsolete with the patch here?

**Comment by BYK** [comment]
Just modified `server.py` instead as we don't seem to be overriding `CACHE_VERSION` in anywhere else.

**Comment by BYK** [comment]
@mattrobenolt now I see the issue. We don't use `VERSION` in `nodedata`. I think I'll need to fix that too :(

**Comment by BYK** [comment]
Split the test refactoring needed from the actual change (mostly). Tests will fail until ##22322 is merged.

**Review by mattrobenolt — APPROVED** [review]

**Review by wedamija — APPROVED** [review]

**Review by evanpurkhiser — APPROVED** [review]
